### PR TITLE
Make Random package not depend on global state

### PR DESCRIPTION
**Pull request by nicolodavis**
#### We have the following problem at the moment:
The `Random` package is bundled into `react.js`, `core.js` and `client.js`. When the user calls the API, they use the version in `core.js`. However, when the framework tries to persist the PRNG state, it uses the version in `react.js` (if using a React client). This loses the "global" state, and makes the Random package always generate the same numbers, regardless of which seed is used.

#### Fix

This PR injects the Random API into `ctx` instead, so you would use it like:

```
moves: {
  rollDie: (G, ctx) => ({ ...G, die: ctx.random.D6() })
}
```

This also has the nice property that the individual move sub-reducers are now pure.

#### Checklist

* [x] Use a separate branch in your local repo (not `master`).
* [x] Test coverage is 100% (or you have a story for why it's ok).

### CODE REVIEW

**Comment by nicolodavis** [comment]
@Stefan-Hanke @sladwig Can you take a look?

File: src/core/reducer.js [hunk 1]
```diff
@@ -84,11 +86,12 @@ export function createGameReducer({ game, numPlayers, multiplayer }) {
         }
 
         // Init PRNG state.
```
Comment (Stefan-Hanke) [inline src/core/reducer.js]: That comment is out of date.

File: src/core/flow.js [hunk 2]
```diff
@@ -278,7 +278,8 @@ export function FlowWithPhases({
   const startTurn = function(state, config) {
     const ctx = { ...state.ctx };
     const G = config.onTurnBegin(state.G, ctx);
-    const _undo = [{ G, ctx }];
+    const { random, ...ctxWithoutAPI } = ctx; // eslint-disable-line no-unused-vars
```
Comment (Stefan-Hanke) [inline src/core/flow.js]: I'd like to hide how `random` is attached to `ctx`. How about a function inside _random.js_ that performs that action?

How does this scale to, say, the Events API? Does this have to be attached/stripped likewise?

**Comment by Stefan-Hanke** [comment]
Looks ok to me.

I had expected you'd need to change the JSON inside _index.test.js_ also, but then, why should it change; the `seed` needs to be stored anyway...

**Comment by sladwig** [comment]
Looks goot to me to, will test it later in game.

File: src/core/flow.js (hunk 2 above)
Comment (nicolodavis) [reply src/core/flow.js]: Done. The Events API will probably have a similar `attach / detach` mechanism for ctx.

File: src/core/reducer.js (hunk 1 above)
Comment (nicolodavis) [reply src/core/reducer.js]: Fixed.

**Pull request merged**